web/src/main/java/org/cbioportal/web/util/StudyViewFilterApplier.java

```diff
@@ -189,138 +183,145 @@ public SampleIdentifier apply(Sample sample) {
         }).collect(Collectors.toList());
     }
 
-    private List<SampleIdentifier> filterMutatedGenes(List<MutationGeneFilter> mutatedGenes, List<SampleIdentifier> sampleIdentifiers) {
-        for (MutationGeneFilter molecularProfileGeneFilter : mutatedGenes) {
-            List<String> studyIds = new ArrayList<>();
-            List<String> sampleIds = new ArrayList<>();
-            List<Integer> entrezGeneIds = geneService
-                    .fetchGenes(molecularProfileGeneFilter.getHugoGeneSymbols(), GeneIdType.HUGO_GENE_SYMBOL.name(), Projection.SUMMARY.name())
-                    .stream()
-                    .map(gene -> gene.getEntrezGeneId())
-                    .collect(Collectors.toList());
-            
-            studyViewFilterUtil.extractStudyAndSampleIds(sampleIdentifiers, studyIds, sampleIds);
-            List<Mutation> mutations = mutationService.getMutationsInMultipleMolecularProfiles(
-                    molecularProfileService.getFirstMutationProfileIds(studyIds, sampleIds), sampleIds, entrezGeneIds,
-                    Projection.ID.name(), null, null, null, null);
-            sampleIdentifiers = mutations.stream().map(m -> {
-                SampleIdentifier sampleIdentifier = new SampleIdentifier();
-                sampleIdentifier.setSampleId(m.getSampleId());
-                sampleIdentifier.setStudyId(m.getStudyId());
-                return sampleIdentifier;
-            }).distinct().collect(Collectors.toList());
-        }
+    private List<SampleIdentifier> filterMutatedOrFusionGenes(List<GeneFilter> genefilters,
```

Comment:
I would suggest keeping filterMutationGenes and filterFusionGenes separate, since the fusion genes table will be dropped in the near future and replaced by paired genes (a gene and its partner gene) from structural variants.

Reply:
you mean separating the logic in this method or separate them in the `StudyViewFilter` object?